akka-stream/src/main/scala/akka/stream/io/Framing.scala

```diff
@@ -0,0 +1,173 @@
+package akka.stream.io
+
+import java.nio.ByteOrder
+
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ TerminationDirective, PushPullStage, SyncDirective, Context }
+import akka.util.{ ByteIterator, ByteStringBuilder, ByteString }
+
+import scala.annotation.tailrec
+
+object Framing {
+
+  def delimiter(delimiterBytes: ByteString, maximumFrameLength: Int): Flow[ByteString, ByteString, Unit] =
```

Comment:
It's just that encoding the type in the name is not generally recommended, e.g. you would not write `maximumFrameLengthInt`, and in another place you have used `maximumLineBytes` which refers to the size in bytes and not the ByteString as type. Naming is hard.